services/registry/handlers/maintainers.js

```diff
@@ -16,88 +10,72 @@ module.exports = [
+    authn.required(accept)
   ),
   fork.del(
     '/v1/packages/package/:namespace([^@]+)@:host/:name/invitation/:member',
-    packageExists(isNamespaceMember(decline))
+    authn.required(decline)
   )
 ];
 
 async function maintainers(context, { namespace, host, name }) {
-  const pkg = await Package.objects
-    .get({
-      active: true,
-      name,
-      'namespace.active': true,
-      'namespace.name': namespace
-    })
-    .catch(Package.objects.NotFound, () => null);
-
-  if (!pkg) {
+  const [err, maintainers] = await context.storageApi.getActiveMaintainers({namespace, host, name, page: 0}).then(
+    xs => [null, xs],
+    xs => [xs, null]
+  )
+
+  if (err) {
+    if (err.status === 404) {
+      return response.error(
+        `"${namespace}@${host}/${name}" does not exist.`,
+        404
+      );
+    }
+
+    context.logger.error(err.message || err)
     return response.error(
-      `"${namespace}@${host}/${name}" does not exist.`,
-      404
+      `Caught error fetching maintainers for "${namespace}@${host}/${name}".`,
+      500
     );
   }
-
-  const namespaces = await Namespace.objects
-    .filter({
-      'maintainers.package_id': pkg.id,
-      'maintainers.accepted': true,
-      'maintainers.active': true
-    })
-    .then();
-
-  const objects = namespaces.map(ns => ns.name).sort();
   return response.json({ objects });
 }
 
 // Invite a maintainer. Maintainers are namespaces, which might be a single user or a group of users.
 // More correctly: maintainership is a relationship between a namespace and a package.
 async function invite(context, { namespace, host, name, invitee }) {
-  if (!context.pkg) {
-    return response.error(`"${namespace}@${host}/${name}" not found.`, 404);
-  }
-
-  if (!context.invitee) {
-    return response.error(`${invitee} not found.`, 404);
-  }
+  const [err, invite] = await context.storageApi.inviteMaintainer({
+    namespace,
+    host,
+    name,
+    from: context.user.name,
+    to: invitee
+  }).then(
+    xs => [null, xs],
+    xs => [xs, null]
+  )
 
-  const existing = await Maintainer.objects
-    .get({
-      namespace: context.invitee,
-      package: context.pkg
-    })
-    .catch(Maintainer.objects.NotFound, () => null);
-  if (existing) {
-    if (existing.active === false) {
-      return response.message(
-        `${invitee} has already declined to maintain ${namespace}@${host}/${name}.`
-      );
-    }
-    if (existing.accepted === false) {
-      return response.message(
-        `${invitee} has already been invited to maintain ${namespace}@${host}/${name}.`
-      );
-    }
+  if (err) {
+    const msg = {
+      'maintainer.invite.invitee_dne': `Unknown namespace: "${invitee}".`,
+      'maintainer.invite.package_dne': `Unknown package: "${invitee}".`,
+      'maintainer.invite.already_accepted': `Namespace "${invitee}" is already a member.`,
+      'maintainer.invite.already_declined': `Namespace "${invitee}" has declined this invite.`
+    }[err.code]
```

Comment:
The error mapping pattern emerges. It's almost worth making a decorator for this and throwing the error, I think. This is where we should hang i18n mapping as well, potentially.